### PR TITLE
JAVA-18149 Upgrade libraries-2 module to JDK 11

### PR DESCRIPTION
**Pull request by timis1**
Migrate libraries-2 to jdk 11

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1074,6 +1072,7 @@
                 <module>tensorflow-java</module>
                 <module>xstream</module>
                 <module>webrtc</module>
+				<module>libraries-2</module>
```
Comment (dkapil) [inline pom.xml:1075]: let's add these modules in alphabetical order. Atleast that is what we are trying to maintain :)

File: libraries-2/src/test/java/com/baeldung/parallel_collectors/ParallelCollectorsUnitTest.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public void shouldProcessInParallelWithParallelCollectors() {
         List<Integer> ids = Arrays.asList(1, 2, 3);
 
         CompletableFuture<List<String>> results = ids.stream()
-          .collect(parallelToList(i -> fetchById(i), executor, 4));
+          .collect(parallelToList(ParallelCollectorsUnitTest::fetchById, executor, 4));
```
Comment (dkapil) [inline libraries-2/src/test/java/com/baeldung/parallel_collectors/ParallelCollectorsUnitTest.java:43]: is this an improvement or a fix?

**Review by dkapil — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (timis1) [reply pom.xml:1069]: I will do

File: libraries-2/src/test/java/com/baeldung/parallel_collectors/ParallelCollectorsUnitTest.java (hunk 2 above)
Comment (timis1) [reply libraries-2/src/test/java/com/baeldung/parallel_collectors/ParallelCollectorsUnitTest.java:43]: Yes this is an improvement to is the new way to call a static method in a lambda expression.

File: pom.xml (hunk 1 above)
Comment (timis1) [reply pom.xml]: Done

File: libraries-2/pom.xml [hunk 3]
```diff
@@ -16,15 +16,7 @@
...
-            <url>http://repository.jboss.org/nexus/content/groups/public/</url>
-            <releases>
-                <enabled>true</enabled>
-                <updatePolicy>never</updatePolicy>
-            </releases>
-            <snapshots>
-                <enabled>true</enabled>
-                <updatePolicy>daily</updatePolicy>
-            </snapshots>
+            <url>https://repo.maven.apache.org/maven2</url>
```
Comment (lor6) [inline libraries-2/pom.xml:19]: do we need to add this repository here explicitly? I think this is the default repository

File: libraries-2/pom.xml [hunk 4]
```diff
@@ -119,22 +117,40 @@
...
+            <version>18.3.12</version>
+        </dependency>
     </dependencies>
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.0.0-M7</version>
```
Comment (lor6) [inline libraries-2/pom.xml:132]: can we use the property from the main pom to define the version here, or is the 3.0.0-M7 required?

File: pom.xml [hunk 5]
```diff
@@ -999,6 +997,7 @@
                 <module>java-native</module>
                 <module>jsoup</module>
                 <module>ksqldb</module>
+				<module>libraries-2</module>
```
Comment (lor6) [inline pom.xml:1000]: let's fix the formatting here and use spaces instead of tabs

File: libraries-2/pom.xml (hunk 3 above)
Comment (timis1) [reply libraries-2/pom.xml:19]: Yes, you are right. I remove the entire block for **<repositories>**.

File: libraries-2/pom.xml (hunk 4 above)
Comment (timis1) [reply libraries-2/pom.xml:132]: Done I added the variable of **maven-surefire-plugin.version**.

File: pom.xml (hunk 5 above)
Comment (timis1) [reply pom.xml:1000]: Done

**Pull request merged**